### PR TITLE
[Week7]-Task3..[Week10]: Task1, Task2

### PR DESCRIPTION
**Pull request by cht8687**
class Functor f where  
    fmap :: (a -> b) -> f a -> f b  

  Functor law:
    fmap id = id
    fmap (g . h) = fmap g . fmap h

Applicative Functor:

 class Functor f => Applicative f where
  pure :: a -> f a
  (<*>) :: f (a -> b) -> f a -> f b

Applicative law:
  pure id <*> x = x
  pure (g x) = pure g <*> pure x
  x <*> pure y = pure (\g -> g y) <*> x
  x <*> (y <*> z) = (pure (.) <*> x <*> y) <*> z

### CODE REVIEW

File: cis194/week7/haotian/week7/src/Scrabble.hs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import Data.Monoid
+import Data.Char
+
+newtype Score = Score Int
+  deriving (Eq, Read, Show, Ord, Num)
+
+instance Monoid Score where
+  mempty  = Score 0
+  mappend = (+)
```
Comment (stevemao) [inline cis194/week7/haotian/week7/src/Scrabble.hs]: This is duplicated definition.

Comment (cht8687) [reply cis194/week7/haotian/week7/src/Scrabble.hs]: explain to me why? I am dumb. 😄 

does it because Monoid belongs to Semigroup?

Comment (stevemao) [reply cis194/week7/haotian/week7/src/Scrabble.hs]: `mappend` is `(<>)` which is already defined below

Comment (cht8687) [reply cis194/week7/haotian/week7/src/Scrabble.hs]: @stevemao great! thanks

**Pull request merged**